core/src/main/java/hudson/model/queue/Executables.java

```diff
@@ -27,19 +27,35 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * Convenience methods around {@link Executable}.
  *
  * @author Kohsuke Kawaguchi
  */
 public class Executables {
+    
+    private static final Logger LOGGER = Logger.getLogger(Executables.class.getName());
+    
+    // TODO: Deprecate getParentOf() and make the new API public
+    // @deprecated This method may throw Runtime exceptions for old cores
+    // Use {@link #getParentOfOrFail(hudson.model.Queue.Executable)} or {@link #getParentOfOrNull(hudson.model.Queue.Executable)} instead.
+    
     /**
      * Due to the return type change in {@link Executable}, the caller needs a special precaution now.
+     * @param e Executable
+     * @return Discovered subtask
+     * @throws Error Executable type does not offer the {@link Executable#getParent()} method or it fails with {@link Error}
+     * @throws RuntimeException {@link Executable#getParent()} method fails with {@link Error}
      */
-    public static @Nonnull SubTask getParentOf(Executable e) {
+    public static @Nonnull SubTask getParentOf(@Nonnull Executable e) 
```

Comment:
I think the correct thing is to just return `e` in case of those issues and remove all the junk. I'd need to confirm on other than my phone though